### PR TITLE
CLI M005/S02 Namespace slice branches for monorepo safety

### PR DESCRIPTION
**Pull request by gannonh**
## What Changed
See slice plan: S02: Namespace slice branches for monorepo safety

## Must-Haves
- R002 (owned): New slice branches are namespaced by project scope and cannot silently reuse stale branches from unrelated projects.
- R002 (owned): Legacy `kata/<M>/<S>` branches remain compatible during transition (parse + checkout continuity).
- R001 (supporting): `worktree.ts` provenance checks consume `runGit()`/`getCurrentBranch()` from `git-service.ts` rather than ad-hoc git shell execution in touched paths.
- Boundary contract closure: `parseBranchToSlice()` + PR context builders operate correctly with namespaced branches and still parse legacy branches.

## Tasks
- T01: Add failing regression tests for namespaced/legacy branch contracts
- T02: Implement namespaced branch generation and provenance guard in worktree layer
- T03: Propagate dual-format parsing and namespaced branch wiring through PR/backends
- T04: Close remaining branch-format drift and prove slice-level integration

<h3>Greptile Summary</h3>

This PR namespaces Kata slice branches from the legacy `kata/<M>/<S>` format to `kata/<scope>/<M>/<S>`, where scope is derived from the project's path relative to the git root. This prevents silent branch reuse across unrelated projects in a monorepo. Legacy branches remain checkout-compatible during transition, and a provenance guard in `ensureSliceBranch` rejects cross-scope branch reuse.

Key changes:
- `worktree.ts`: `getSliceBranchName` gains a `basePath` argument; `getProjectScope` derives the scope segment from the relative path; `ensureSliceBranch` gains a provenance check (`findConflictingNamespacedBranch`) that throws when a different-scope branch exists for the same milestone/slice; `parseSliceBranchName` is exported as a dual-format parser.
- `file-backend.ts` / `linear-backend.ts`: Both `preparePrContext` implementations replace the hardcoded `kata/<M>/<S>` string with a dynamic resolution via `getCurrentBranch` after `ensureSliceBranch`, preserving the legacy branch name during transition.
- `gh-utils.ts`: `parseBranchToSlice` updated to accept both branch formats (namespaced first, then legacy fallback).
- Tests: New regression tests for namespaced branch contracts in all four test files; new `gh-utils.test.ts` covers `parseBranchToSlice` format coverage.

**Issues found:**
- The provenance guard in `ensureSliceBranch` has a bypass path: when a legacy branch (`kata/M001/S05`) coexists alongside a conflicting namespaced branch (`kata/other-scope/M001/S05`), `resolvePreferredSliceBranch` returns the legacy branch before `findConflictingNamespacedBranch` is ever called. The conflict is silently ignored. This case is not covered by any test.
- `mergeSliceToMain` falls back to the legacy branch without a scope check, so callers invoking it directly (without `ensureSliceBranch`) skip provenance validation.
- `parseBranchToSlice` in `gh-utils.ts` and `parseSliceBranchName` in `worktree.ts` are duplicate implementations of the same dual-format parsing logic.

<h3>Confidence Score: 3/5</h3>

- Merge with caution — the core namespacing logic is sound but the provenance guard has an unverified bypass in the legacy-branch coexistence scenario.
- The branch naming, scope derivation, and dual-format parsing are well-implemented and thoroughly tested for the happy path. The provenance check correctly blocks cross-scope reuse when starting from scratch. However, the guard is silently bypassed when both a legacy branch and a conflicting namespaced branch exist simultaneously, which is exactly the monorepo transition scenario the PR aims to protect against. Additionally, `mergeSliceToMain` has no scope validation of its own. These gaps reduce confidence that R002 is fully closed.
- `apps/cli/src/resources/extensions/kata/worktree.ts` — specifically the `ensureSliceBranch` provenance logic (lines 149–202) and `mergeSliceToMain` legacy fallback (lines 236–240).

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/cli/src/resources/extensions/kata/worktree.ts | Core branch management rewritten to support namespaced `kata/<scope>/<M>/<S>` format alongside legacy `kata/<M>/<S>`; introduces provenance guard that has a bypass path when a legacy branch coexists with a conflicting namespaced branch, and `mergeSliceToMain` falls back to legacy branch without a scope check. |
| apps/cli/src/resources/extensions/kata/file-backend.ts | Replaces hardcoded `kata/<M>/<S>` branch string with dynamic resolution via `getCurrentBranch`; the `startsWith("kata/")` guard is over-broad but functionally harmless post-`ensureSliceBranch`. |
| apps/cli/src/resources/extensions/kata/linear-backend.ts | Migrates `preparePrContext` from ad-hoc `execSync` git calls to `ensureSliceBranch`/`getCurrentBranch` from `worktree.ts`; same over-broad `startsWith("kata/")` pattern as file-backend. |
| apps/cli/src/resources/extensions/pr-lifecycle/gh-utils.ts | `parseBranchToSlice` updated to accept both namespaced and legacy formats; duplicate parsing logic mirrors `parseSliceBranchName` in `worktree.ts`, creating a two-implementation maintenance risk. |
| apps/cli/src/resources/extensions/kata/tests/worktree.test.ts | Good coverage of namespaced branch creation, idempotency, legacy continuity, and provenance rejection; missing the edge case where both a legacy and a conflicting namespaced branch coexist simultaneously. |
| apps/cli/src/resources/extensions/pr-lifecycle/tests/gh-utils.test.ts | New test file with solid coverage of `parseBranchToSlice` for namespaced, legacy, null, and malformed inputs. |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[ensureSliceBranch called] --> B{Already on namespaced\nor legacy branch?}
    B -- Yes --> C[return false\nno-op]
    B -- No --> D[resolvePreferredSliceBranch]
    D --> E{Namespaced branch\nkata/scope/M/S exists?}
    E -- Yes --> F[checkout namespaced branch\nreturn false]
    E -- No --> G{Legacy branch\nkata/M/S exists?}
    G -- Yes --> H[checkout legacy branch\nreturn false]
    G -- No --> I[findConflictingNamespacedBranch]
    I --> J{Other-scope branch\nfor same M/S found?}
    J -- Yes --> K[throw Error\nprovenance mismatch]
    J -- No --> L[create kata/scope/M/S\nfrom main]
    L --> M[checkout new namespaced branch\nreturn true]

    style H fill:#ffdd99,stroke:#cc8800
    style K fill:#ffcccc,stroke:#cc0000
    style F fill:#ccffcc,stroke:#006600
    style M fill:#ccffcc,stroke:#006600

    N[⚠️ Bypass gap: if both G and J\nare true simultaneously,\nconflicting branch is silently ignored]
    H -.-> N
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/cli/src/resources/extensions/kata/worktree.ts
Line: 149-183

Comment:
**Provenance guard bypassed when legacy and conflicting branches coexist**

`resolvePreferredSliceBranch` returns the legacy branch (`kata/M001/S05`) whenever it exists, which means `findConflictingNamespacedBranch` is never reached in that code path. Consider this scenario in a monorepo:

- `kata/project-b/M001/S05` exists (committed by another project)
- `kata/M001/S05` also exists (a legacy branch)

When `ensureSliceBranch` is called for project-a (scope = `project-a`):
1. `resolvePreferredSliceBranch` checks `kata/project-a/M001/S05` → not found
2. Checks `kata/M001/S05` → found → returns it immediately
3. Checks out the legacy branch, returns `false`
4. `findConflictingNamespacedBranch` is **never called** — `kata/project-b/M001/S05` conflict is silently ignored

The R002 requirement ("cannot silently reuse stale branches from unrelated projects") is only enforced when no legacy branch exists alongside a conflicting namespaced branch. The provenance check should run before falling back to the legacy branch, or at least emit a warning when a conflicting namespaced branch coexists with the chosen legacy branch.

The test coverage in `worktree.test.ts` only covers the case where there is no legacy branch (`kata/M001/S05` is absent), leaving this bypass path untested.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/cli/src/resources/extensions/kata/file-backend.ts
Line: 384-391

Comment:
**Overly broad branch detection after `ensureSliceBranch`**

The `startsWith("kata/")` check accepts any kata branch as `branch`, not specifically the one that `ensureSliceBranch` just established. The function is documented to ensure the correct branch, so using the return value of `getCurrentBranch` unconditionally would be both simpler and more accurate. The fallback to `getSliceBranchName` can only be reached if the caller was on a non-kata branch and `ensureSliceBranch` somehow did not switch — which would indicate a bug that this code would then mask.

The same pattern appears in `linear-backend.ts` at the equivalent `preparePrContext` method (diff lines ~55–58).

```suggestion
    const currentBranch = getCurrentBranch(this.basePath);
    const branch = currentBranch;
```

If the fallback is intentional for detached-HEAD scenarios, a comment explaining the rationale would help readers understand why `startsWith("kata/")` is used instead of the branch name returned by `ensureSliceBranch`.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/cli/src/resources/extensions/kata/worktree.ts
Line: 290-316

Comment:
**Duplicate branch-parsing logic between `worktree.ts` and `gh-utils.ts`**

`parseSliceBranchName` (exported here) and `parseBranchToSlice` in `pr-lifecycle/gh-utils.ts` both parse the same two branch formats (`kata/<scope>/<M>/<S>` and `kata/<M>/<S>`) with near-identical regex patterns. Having two implementations means any future format change must be updated in two places and both test suites must be kept in sync.

`gh-utils.ts` is in a different package (`pr-lifecycle`), but since it already has a known cross-extension dependency path, it could import `parseSliceBranchName` from `worktree.js` and derive `{ milestoneId, sliceId }` from the result, keeping the parsing logic in a single source of truth.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/cli/src/resources/extensions/kata/worktree.ts
Line: 236-240

Comment:
**`mergeSliceToMain` silently falls back to legacy branch without provenance check**

When neither the namespaced branch nor the legacy branch exists, the function sets `branch = legacyBranch` and then throws at the `!branchExists` guard below — that's fine. But if the namespaced branch doesn't exist while the legacy branch does, the code merges the legacy branch without verifying it belongs to this project scope. In a shared monorepo, `kata/M001/S01` might have been created by a different project, and its history would be merged into this project's main branch without any warning.

The `ensureSliceBranch` provenance guard is not called before `mergeSliceToMain`, so callers that call `mergeSliceToMain` directly (without going through `ensureSliceBranch` first) bypass scope validation entirely. A scope check here, or at minimum a comment noting that the caller is responsible for provenance validation, would improve safety.

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Last reviewed commit: ["feat(S02/T04): align..."](https://github.com/gannonh/kata/commit/193396277f9ac0e0179a43b9f5d4efa86f36a90e)</sub>

> Greptile also left **4 inline comments** on this PR.

## Summary by CodeRabbit

* **New Features**
  * Support namespaced slice branches (kata/<scope>/Mxxx/Syy) alongside legacy format; PR title normalization and per-step model preference added.
  * Added short prompt snippets to many tool descriptors and a new Linear "add comment" action.

* **Behavior Changes**
  * PR creation may push the current branch if missing and provides clearer branch-format errors/hints; auto-mode status shows model info and warns if preferred model is unavailable.

* **Tests**
  * Added/updated tests covering namespaced vs legacy branch parsing and Git-backed workflows.

### CODE REVIEW

**Comment by gannonh** [comment]
@coderabbit review

File: apps/cli/src/resources/extensions/kata/worktree.ts [hunk 1]
```diff
@@ -11,83 +11,205 @@
...
 
-  if (current === branch) return false;
+  if (current === namespacedBranch || current === legacyBranch) return false;
 
-  const mainBranch = getMainBranch(basePath);
-  let created = false;
+  const preferredExisting = resolvePreferredSliceBranch(basePath, milestoneId, sliceId);
+  if (preferredExisting) {
+    runGit(basePath, ["checkout", preferredExisting]);
+    return false;
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/cli/src/resources/extensions/kata/worktree.ts]: Fixed. Removed `|| current === legacyBranch` from the fast-path so `resolvePreferredSliceBranch()` (which already contains the conflict check) always runs when on a legacy branch. The namespaced fast-path remains since it's unambiguous.

File: apps/cli/src/resources/extensions/linear/linear-tools.ts [hunk 2]
```diff
@@ -98,6 +99,7 @@ export function registerLinearTools(pi: ExtensionAPI, client: LinearClient) {
     name: "linear_get_team",
     label: "Linear: Get Team",
     description: "Get a team by key (e.g. 'KAT') or UUID.",
+    promptSnippet: "Get a team by key (e.g.",
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/cli/src/resources/extensions/linear/linear-tools.ts]: Fixed. Truncated snippets cleaned up — now read `"Get a team by key or UUID."` and `"Get an issue by UUID or identifier."` instead of cut-off sentences.

File: apps/cli/src/resources/extensions/mac-tools/index.ts [hunk 3]
```diff
@@ -391,6 +397,7 @@ export default function (pi: ExtensionAPI) {
 			"- 'tree': Dump the full accessibility subtree as an indented tree. Use maxDepth/maxCount to bound output.\n" +
 			"- 'focused': Get the currently focused element in the app. No criteria needed.\n" +
 			"The 'app' param accepts an app name (e.g. 'Finder') or bundle ID (e.g. 'com.apple.Finder').",
+		promptSnippet: "Find UI elements in a macOS applications accessibility tree.",
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/cli/src/resources/extensions/mac-tools/index.ts]: Fixed. Both occurrences now use the possessive: `"macOS application's accessibility tree"` and `"macOS application's UI structure"`.

File: apps/cli/src/resources/extensions/pr-lifecycle/index.ts [hunk 4]
```diff
@@ -449,6 +453,7 @@ export default function (pi: ExtensionAPI): void {
       "or { ok: false, phase, error } on failure.",
       "Phase enum: gh-missing | gh-unauth | reply-failed.",
     ].join(" "),
+    promptSnippet: "Replies to an inline GitHub PR review thread via the",
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/cli/src/resources/extensions/pr-lifecycle/index.ts]: Fixed. Now reads `"Reply to an inline GitHub PR review thread."` — complete sentence.

File: apps/cli/src/resources/extensions/pr-lifecycle/index.ts [hunk 5]
```diff
@@ -582,9 +588,11 @@ export default function (pi: ExtensionAPI): void {
         return toolFail({
           ok: false,
           phase: "branch-parse-failed",
-          error: `Current branch '${branch}' does not match kata/<MilestoneId>/<SliceId> pattern`,
+          error:
+            `Current branch '${branch}' does not match supported Kata slice branch formats: `
+            + "kata/<scope>/<MilestoneId>/<SliceId> or legacy kata/<MilestoneId>/<SliceId>",
           hint:
-            "Switch to a Kata slice branch (e.g. kata/M003/S04) or pass milestoneId and sliceId explicitly.",
+            "Switch to a Kata slice branch (e.g. kata/apps-cli/M003/S04, or legacy kata/M003/S04 during transition) or pass milestoneId and sliceId explicitly.",
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/cli/src/resources/extensions/pr-lifecycle/index.ts]: Good catch. Fixed — hint now says `"before running kata_merge_pr"` instead of referencing `milestoneId`/`sliceId` params that don't exist on this tool.

File: apps/cli/src/resources/extensions/pr-lifecycle/pr-runner.ts [hunk 6]
```diff
@@ -189,13 +191,50 @@ export async function runCreatePr(options: PrCreateOptions): Promise<PrCreateRes
...
+          } catch (pushErr) {
+            const pe = pushErr as { stderr?: string; message?: string };
+            return {
+              ok: false,
+              phase: "push-failed",
+              error: `Branch '${head}' is not on the remote and push failed: ${pe.stderr ?? pe.message ?? String(pushErr)}`,
+              hint: "Check git remote configuration and network connectivity. Run: git remote -v",
+            };
+          }
+        }
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/cli/src/resources/extensions/pr-lifecycle/pr-runner.ts:220]: Disagree with the "critical" severity. The `head` variable comes from `git rev-parse --abbrev-ref HEAD` which returns sanitized branch names — not arbitrary user input. The shell injection concern doesn't apply here.

On the stale-commits point: when the remote ref already exists, the branch was pushed either by `preparePrContext` (which runs `git push -u origin` immediately before) or by a previous PR attempt. The pre-push hook runs the full validation gate on every push. Comparing SHAs and force-pushing on every `kata_create_pr` call would add latency and complexity for a scenario that doesn't occur in practice — the push guard is specifically for the "branch never pushed" case (KAT-635).
